utils/dependencies_test.go

```diff
@@ -0,0 +1,55 @@
+// +build unit
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package utils
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestLocationIsGithub(t *testing.T) {
+	assert.True(t, LocationIsGithub("github.com/my-org/my-project"), "Does not allow github without a http/https prefix")
+	assert.True(t, LocationIsGithub("github.ibm.com/my-org/my-project"), "Does not allow github.ibm.com without a http/https prefix")
+	assert.True(t, LocationIsGithub("http://github.com/my-org/my-project"), "Does not allow github with a http/https prefix")
+	assert.True(t, LocationIsGithub("http://github.ibm.com/my-org/my-project"), "Does not allow github.ibm.com with a http/https prefix")
+}
+
+func TestLocationIsGithub_WithUsernamePassword(t *testing.T) {
+	assert.True(t, LocationIsGithub("username:password@github.com/my-org/my-project"), "Does not allow username/password and github without a http/https prefix")
```

Comment:
This is new functionality... the working assumption for wskdeploy was it would work with public repos. only.  So this PR is a feature request.

Reply:
Perhaps we should return more info from the function like the uname:pwd and flag indicating public/private on the result set for the caller to use.

Reply:
You can see I took the path of least resistance with this PR. I am happy to hear what a proper solution is from you all and adjust this PR as needed.

Reply:
Thanks @allen-servedio for initiating this effort. Your PR looks great except that these changes creates perception that `wskdeploy` supports private GitHub repos, which is not true.  Many times, people look for examples and just `grep` for specific keywords like `username:password@github.com. Can you please add a comment in the test file saying that we still do not have support for private repo but this is one of many changes for adding that support and we can get rid of such comment once its supported. Thoughts?

Reply:
Will do.